app/models/user.rb

```diff
@@ -0,0 +1,37 @@
+class User < ActiveRecord::Base
+  validates :email,    presence: { message: "Must have an email." },
+                       uniqueness: true
+  validates :password, length:   { in: 6..20 },
+                       allow_nil: true
+  has_secure_password
+
+  has_many :posts
+  has_many :likes
+  has_many :comments
+
+  has_one :profile
```

Comment:
Not sure if this was intended, but maybe you want to destroy a user's profile when a user is destroyed